sema/subyt/sinks.py

```
@@ -9,17 +9,18 @@
-    if not os.path.exists(parent_path):
-        os.makedirs(parent_path)
-    assert os.access(
-        parent_path, os.W_OK
-    ), f"Can not write to folder '{parent_path}' for creating new files"
+        assert not out_path.exists(), (
+            f"File to write '{path_name}' already exists",
+        )
+    # ensure parent folder exists
+    parent_path = out_path.parent.absolute()
+    parent_path.mkdir(parents=True, exist_ok=True)
+    assert os.access(parent_path, os.W_OK), (
```

Ruff (0.8.2)

Use of `assert` detected

(S101)